src/egg_hatch.c

```diff
@@ -50,3 +269,596 @@ void CreatedHatchedMon(struct Pokemon *egg, struct Pokemon *temp) {
+            offsetWrite2 = (u32)(VRAM + 0x6000);
+            DmaCopy16(3, offsetRead2, (void*)(offsetWrite2), 0x1000);
+            gMain.state++;
+        }
+        break;
+    case 8:
+        REG_BG1CNT = 0x501;
+
+        REG_BG0HOFS = 0;
+        REG_BG0VOFS = 0;
+
+        REG_BG1HOFS = 0;
+        REG_BG1VOFS = 0;
+
+        REG_BG2HOFS = 0;
+        REG_BG2VOFS = 0;
+
+        SetMainCallback2(CB2_EggHatch_1);
+        gEggHatchData->CB2_state = 0;
+        break;
+    }
+}
+
+static void EggHatchSetMonNickname(void)
+{
+    SetMonData(&gPlayerParty[gSpecialVar_0x8004], MON_DATA_NICKNAME, gStringVar3);
+    SetMainCallback2(c2_exit_to_overworld_2_switch);
+}
+
+static void Task_EggHatchPlayBGM(u8 taskID)
+{
+    if (gTasks[taskID].data[0] == 0)
+        StopMapMusic();
+    if (gTasks[taskID].data[0] == 1)
+        PlayBGM(376);
+    if (gTasks[taskID].data[0] > 60)
+    {
+        PlayBGM(377);
+        DestroyTask(taskID);
+        //return; task is destroyed, yet you increment the value?
+    }
+    gTasks[taskID].data[0]++;
+}
+
+static void CB2_EggHatch_1(void)
+{
+    switch (gEggHatchData->CB2_state)
+    {
+    case 0:
+        BeginNormalPaletteFade(-1, 0, 0x10, 0, 0);
+        REG_DISPCNT = 0x1740;
+        gEggHatchData->CB2_state++;
+        CreateTask(Task_EggHatchPlayBGM, 5);
+        break;
+    case 1:
+        if (!gPaletteFade.active)
+        {
+            gEggHatchData->CB2_PalCounter = 0;
+            gEggHatchData->CB2_state++;
+        }
+        break;
+    case 2:
+        if (++gEggHatchData->CB2_PalCounter > 30)
+        {
+            gEggHatchData->CB2_state++;
+            gSprites[gEggHatchData->eggSpriteID].callback = SpriteCB_Egg_0;
+        }
+        break;
+    case 3:
+        if (gSprites[gEggHatchData->eggSpriteID].callback == SpriteCallbackDummy)
+            gEggHatchData->CB2_state++;
+        break;
+    case 4:
+        GetMonNick(&gPlayerParty[gEggHatchData->eggPartyID], gStringVar1);
+        StringExpandPlaceholders(gStringVar4, gOtherText_HatchedFromEgg);
+        EggHatchPrintMessage2(gStringVar4);
+        PlayFanfare(371);
+        gEggHatchData->CB2_state++;
+        break;
+    case 5:
+        if (IsFanfareTaskInactive())
+            gEggHatchData->CB2_state++;
+        break;
+    case 6:
+        if (IsFanfareTaskInactive())
+            gEggHatchData->CB2_state++;
+        break;
+    case 7:
+        GetMonNick(&gPlayerParty[gEggHatchData->eggPartyID], gStringVar1);
+        StringExpandPlaceholders(gStringVar4, gOtherText_NickHatchPrompt);
+        EggHatchPrintMessage1(gStringVar4);
+        gEggHatchData->CB2_state++;
+        break;
+    case 8:
+        if (EggHatchUpdateWindowText())
+        {
+            MenuDrawTextWindow(22, 8, 27, 13);
+            InitYesNoMenu(22, 8, 4);
+            gEggHatchData->CB2_state++;
+        }
+        break;
+    case 9:
+        {
+            s8 menuInput;
+            if ((menuInput = ProcessMenuInputNoWrap_()) != -2)
```

Comment:
This matches, if you think it's more readable.
`            if ((menuInput = ProcessMenuInputNoWrap_()) != -2)
            {
                if (menuInput == -1 || menuInput == 1)
                {
                    gEggHatchData->CB2_state++;
                }
                else
                {
                    u16 species;
                    u8 gender;
                    u32 personality;

                    GetMonNick(&gPlayerParty[gEggHatchData->eggPartyID], gStringVar3);
                    species = GetMonData(&gPlayerParty[gEggHatchData->eggPartyID], MON_DATA_SPECIES);
                    gender = GetMonGender(&gPlayerParty[gEggHatchData->eggPartyID]);
                    personality = GetMonData(&gPlayerParty[gEggHatchData->eggPartyID], MON_DATA_PERSONALITY, 0);
                    DoNamingScreen(3, gStringVar3, species, gender, personality, EggHatchSetMonNickname);
                }
            }`